pkg/azure/gw-machineset.go

```diff
@@ -0,0 +1,83 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package azure
+
+var machineSetYAML = `apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: {{.InfraID}}
+    machine.openshift.io/cluster-api-machine-role: worker
+    machine.openshift.io/cluster-api-machine-type: worker
+  name: {{.InfraID}}-submariner-gw-{{.Region}}{{.AZ}}
+  namespace: openshift-machine-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: {{.InfraID}}
+      machine.openshift.io/cluster-api-machineset: {{.InfraID}}-submariner-gw-{{.Region}}{{.AZ}}
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: {{.InfraID}}
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: {{.InfraID}}-submariner-gw-{{.Region}}{{.AZ}}
+    spec:
+      metadata:
+        labels:
+          submariner.io/gateway: "true"
+      taints:
+        - effect: NoSchedule
+          key: node-role.submariner.io/gateway
+      providerSpec:
+        value:
+          apiVersion: azureproviderconfig.openshift.io/v1beta1
+          credentialsSecret:
+            name: azure-cloud-credentials
+            namespace: openshift-machine-api
+          image:
+            offer: ""
+            publisher: ""
+            resourceID: >-
+              /resourceGroups/{{.InfraID}}-rg/providers/Microsoft.Compute/images/{{.InfraID}}
+            sku: ""
+            version: ""
+          internalLoadBalancer: ""
+          kind: AzureMachineProviderSpec
+          location: {{.Region}}
+          managedIdentity: {{.InfraID}}-identity
+          natRule: null
+          networkResourceGroup: {{.InfraID}}-rg
```

Comment:
```suggestion
          networkResourceGroup: {{.BaseGroupName}}
```
BaseGroupName is part of azureCloud https://github.com/submariner-io/cloud-prepare/pull/203/files#diff-f2027c3046b53bdf075254bbf73d790797640745c6d58bff13b71992570287feR38

Reply:
Applies everywhere.